src/denoising_model.jl

```diff
@@ -0,0 +1,34 @@
+export denoising_model
+
+include("denoising_data.jl")
+
+function denoising_model(shape, shape_p, KERNEL_SIZE, KERNEL_SIGMA = 1.5)
+  sigma = 10^-3
+  data_path = joinpath(@__DIR__, "..", "images/cameraman.png")
+  cameraman_image = Images.load(data_path)
+  x_t = vec(Float64.(cameraman_image))
+  H, H_T, W, W_T = generate_gaussian_blur(shape, shape_p, KERNEL_SIZE, KERNEL_SIGMA)
+  (n, m) = shape
+  b = H(x_t) + randn(n * m) * sigma
+  y = similar(b)
+  z = similar(b)
+
+  function obj(x)
+    y .= W_T(x)
+    y .= H(y)
```

Comment:
Are there in-place versions of `H` and `W_T`? Currently, there are lots of allocations here.